bn_mp_prime_strong_lucas_selfridge.c

```diff
@@ -177,26 +177,12 @@ mp_err mp_prime_strong_lucas_selfridge(const mp_int *a, mp_bool *result)
    mp_set(&U2mz, 1uL);  /* U_1 */
    mp_set(&V2mz, (mp_digit)P);  /* V_1 */
 
-   if (Q < 0) {
-      Q = -Q;
-      mp_set_u32(&Qmz, (uint32_t)Q);
-      if ((err = mp_mul_2(&Qmz, &Q2mz)) != MP_OKAY) {
-         goto LBL_LS_ERR;
-      }
-      /* Initializes calculation of Q^d */
-      mp_set_u32(&Qkdz, (uint32_t)Q);
-      Qmz.sign = MP_NEG;
-      Q2mz.sign = MP_NEG;
```

Comment:
What is with Q2mz here? It is negative afterwards.

Reply:
(Ah, somebody's looking, thanks!)

Yes, that is correct. See the original implementation (file `trn.c`, lines 3145 f.):

```
mpz_set_si(mpzQm, lQ);
mpz_set_si(mpz2Qm, 2*lQ);
```

And because we had no signed `mp_set_*` before I had to go to some lengths which are not necessary anymore.

Reply:
Ah I see now, I confused something. Looks good :+1: